test/odbc/toby_tinyint_pg_datatype.cpp

```diff
@@ -0,0 +1,585 @@
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < operations_query.size(); j++) {
+
+      ASSERT_EQ(col_len[j], BYTES_EXPECTED);
+      ASSERT_EQ(col_results[j], expected_results[i][j]);
+    }
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_int, View_Creation) {
+
+  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
+
+  const int BYTES_EXPECTED = 1;
+
+  unsigned char pk;
+  unsigned char data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector <string> valid_inserted_values = {
+    "0",
+    "255",
+    "3",
+    "NULL"
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
+    {1, SQL_C_STINYINT, &pk, 0, &pk_len},
+    {2, SQL_C_STINYINT, &data, 0,  &data_len}
+  };
+
+  string insert_string{}; 
+  string comma{};
+  
+  for (int i = 0; i< valid_inserted_values.size(); ++i) {
+    insert_string += comma + "(" + std::to_string(i) + "," + valid_inserted_values[i] + ")";
+    comma = ",";
+  }
+
+  // Create Table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS_TINYINT));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+ 
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, valid_inserted_values.size());
+  
+  odbcHandler.CloseStmt();
+
+  // Create view
+  odbcHandler.ExecQuery(CreateViewStatement(VIEW_NAME, VIEW_QUERY));
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(VIEW_NAME, {"*"}, vector<string> {COL1_NAME}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < valid_inserted_values.size(); ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (valid_inserted_values[i] != "NULL")
+    {
+      ASSERT_EQ(data_len, BYTES_EXPECTED);
+      ASSERT_EQ(data, StringToTinyInt(valid_inserted_values[i]));
+    }
+    else 
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
```

Comment:
Is it possible to add some more testing, where we use some standard postgres mathematical operators, e.g.
`select abs(col2), log(col2), power(col2,col1) from TABLE_NAME;`

A list of PG mathematical operators can be found here: https://www.postgresql.org/docs/8.4/functions-math.html#FUNCTIONS-MATH-FUNC-TABLE

We don't have to test all of them, of course.